docs/cookbook.md

```diff
@@ -3,6 +3,54 @@ title:  Ion Schema Language Cookbook
 ---
 # {{ page.title }}
 
+## Optionally ignoring the occurs requirement for fields
+
+Suppose you have a type definition for a struct, and sometimes you want to validate the fields without enforcing the
+required-ness of fields. (The most likely use case for this is to perform partial validation on a partially constructed 
+piece of data.) Ion Schema does not provide any flag or option to "turn off" the required-ness of fields, but you can
+model this in Ion Schema by putting the types of fields and the required-ness of fields in separate type definitions.
```

Comment:
Do we want to call this technique "inheritance"?

Reply:
I think it would be nice to have a name, but I don't think we should use "inheritance" because of the association it has with the additive style of inheritance of OOP. What you might call "inheritance" in ISL (ie. using the `type:` constraint) is subtractive. When one type inherits from another, the set of values that it will accept will always be a subset of the type that it inherits from.

There have been a few times when I was helping someone figure out a schema problem, and it turned out that they thought they could use OOP style inheritance and as a result they had created an unsatisfiable schema.